helix-core/src/comment.rs

```diff
@@ -29,47 +30,53 @@ fn find_line_comment(
                 // considered uncommented.
                 commented = false;
             }
-        } else {
-            // blank line
-            skipped.push(line);
+
+            // determine margin of 0 or 1 for uncommenting; if any comment token is not followed by a space,
+            // a margin of 0 is used for all lines.
+            if matches!(line_slice.get_char(pos + token_len), Some(c) if c != ' ') {
```

Comment:
The `matches!()` here is unneeded since it's only checking a single case.  You can just do:

```rust
if line_slice.get_char(pos + token_len) == Some(' ') {
```